\documentclass[final]{beamer}
\usepackage{grffile}
\mode<presentation>{\usetheme{CambridgeUSPOL}}

\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{graphicx}
\usepackage{array,booktabs,tabularx}
\usepackage{epstopdf}
\usepackage{colortbl, xcolor}
\newcolumntype{Z}{>{\centering\arraybackslash}X}

% rysunki
\usepackage{tikz}
\usepackage{ifthen}
\usepackage{xxcolor}
\usetikzlibrary{arrows}
\usetikzlibrary[topaths]
\usetikzlibrary{decorations.pathreplacing}
%\usepackage{times}\usefonttheme{professionalfonts}  % times is obsolete
\usefonttheme[onlymath]{serif}
\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                       % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

\usecolortheme{seagull}
\useinnertheme{rectangles}
\setbeamercolor{item projected}{bg=darkred}
% \setbeamertemplate{enumerate items}[default]
\setbeamertemplate{caption}{\insertcaption} 
\setbeamertemplate{navigation symbols}{}
\setbeamercovered{transparent}
\setbeamercolor{block title}{fg=darkred}
\setbeamercolor{local structure}{fg=darkred}

\setbeamercolor*{enumerate item}{fg=darkred}
\setbeamercolor*{enumerate subitem}{fg=darkred}
\setbeamercolor*{enumerate subsubitem}{fg=darkred}

%\setbeamercolor*{itemize item}{fg=darkred}
%\setbeamercolor*{itemize subitem}{fg=darkred}
%\setbeamercolor*{itemize subsubitem}{fg=darkred}

\newlength{\columnheight}
\setlength{\columnheight}{94cm}
\renewcommand{\thetable}{}
\def\andname{,}
\authornote{}

\renewcommand{\APACrefatitle}[2]{}
\renewcommand{\bibliographytypesize}{\footnotesize} 
\renewcommand{\APACrefYearMonthDay}[3]{%
  {\BBOP}{#1}
  {\BBCP}
}

\begin{document}

\date{}

\author{\large \underline{Jaros\l{}aw Chilimoniuk}\inst{1}, Pawe\l{} Mackiewicz\inst{1} and Micha\l{} Burdukiewicz\inst{2}*\\
\bigskip
\normaltext{*michal.burdukiewicz@gmail.com}}

\institute{\small{\textsuperscript{1}University of Wroc\l{}aw, Department of Genomics, Wroc\l{}aw, POLAND
\textsuperscript{2}Warsaw University of Technology, Warsaw, POLAND}}

\title{\huge CsgA, CsgB and CsgC - evolutionary interplay in curli biogenesis}

\begin{frame}
\begin{columns}
\begin{column}{.50\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}
\parbox[t][\columnheight]{\textwidth}
{
\begin{block}{Introduction}

<<knitrIntro, echo = FALSE, message=FALSE,warning=FALSE>>=

library(dplyr)
library(ggplot2)
library(reshape2)
library(ggseqlogo)
library(ggrepel)
library(cowplot)
@
  

CsgA, also known as a major curli component, is a secreted protein ubiquitous in biofilms of gram-negative bacteria (Zhou et al., 2012a). Thanks to its ability to create durable fibers, CsgA is a dominant proteinaceous scaffold of biofilms. In fact, CsgA belongs to amyloids, proteins that form fibers during a spontaneous aggregation. CsgC, a sequence-specific inhibitor, is able to arrest aggregation of CsgA in periplasmic space (Evans et al., 2015).

The presence of pre-formed amyloid fibers can accelerate aggregation of other amyloids. This process is known as cross-seeding. It is extremely sequence specific and can be restricted by a difference in a single amino acid (Zhou et al., 2012b). CsgA can be in vivo cross-seeded by its nucleator protein, CsgB, but also other CsgA fibrils.
\end{block}


  \begin{block}{Methods}
  We recently published AmyloGram (Burdukiewicz et al., 2017). It is an efficient tool for prediction of amyloid proteins as CsgA and CsgB. Among others, our algorithm produces a simplified amino acid alphabet for amyloid proteins. It groups amino acid together into six groups based on their physicochemical features. We combined sequences of 2678 triplets CsgA-CsgB-CsgC from closely related gram-negative bacteria and used EVmutation software (Hopf et al., 2017) to find out which residues are especially susceptible to simultaneous mutations. 
\end{block}


  \begin{block}{Results}


<<echo = FALSE,message=FALSE,fig.align='center',fig.width=14,fig.height=10,warning=FALSE>>=

if(Sys.info()[["nodename"]] %in% c("amyloid", "lori")) {
  seq_path_CsgA <- "/home/michal/Dropbox/dropbox-amylogram/PSI-blast/CsgA_muscle.fas"
  seq_path_CsgB <- "/home/michal/Dropbox/dropbox-amylogram/PSI-blast/CsgB_muscle.fas"
}
if(Sys.info()[["nodename"]] %in% c("MSI-GE60-2PE")) {
seq_path_CsgA <- "/home/jarek/Dropbox/amyloids/PSI-blast/CsgA_muscle.fas"
seq_path_CsgB <- "/home/jarek/Dropbox/amyloids/PSI-blast/CsgB_muscle.fas"
}

CsgA_regions <- list(R1 = 43L:65, 
                     R2 = 66L:87,
                     R3 = 88L:110,
                     R4 = 111L:132,
                     R5 = 133L:151)

CsgB_regions <- list(R1 = 45L:66, 
                     R2 = 67L:88,
                     R3 = 89L:110,
                     R4 = 111L:132,
                     R5 = 133L:154)

plot_logo_region <- function(seq_path, region){
  
  all_lines <- readLines(seq_path)
  
  prot_id <- cumsum(grepl("^>", all_lines))
  
  all_prots <- split(all_lines, prot_id)
  
  aln_dat <- lapply(all_prots, function(ith_prot) {
    strsplit(paste0(ith_prot[-1], collapse = ""), "")[[1]]
  }) %>% 
    do.call(rbind, .)
  
  real_positions <- cumsum(aln_dat[1, ] != "-")
  
  ggplot() + geom_logo( apply(aln_dat[, real_positions %in% region], 1, paste0, collapse = "")) + theme_logo()
}


A1 <- plot_logo_region(seq_path = seq_path_CsgA, region = CsgA_regions[["R1"]])
A2 <- plot_logo_region(seq_path = seq_path_CsgA, region = CsgA_regions[["R2"]])
A3 <- plot_logo_region(seq_path = seq_path_CsgA, region = CsgA_regions[["R3"]])
A4 <- plot_logo_region(seq_path = seq_path_CsgA, region = CsgA_regions[["R4"]])
A5 <- plot_logo_region(seq_path = seq_path_CsgA, region = CsgA_regions[["R5"]])


B1 <- plot_logo_region(seq_path = seq_path_CsgB, region = CsgB_regions[["R1"]])
B2 <- plot_logo_region(seq_path = seq_path_CsgB, region = CsgB_regions[["R2"]])
B3 <- plot_logo_region(seq_path = seq_path_CsgB, region = CsgB_regions[["R3"]])
B4 <- plot_logo_region(seq_path = seq_path_CsgB, region = CsgB_regions[["R4"]])
B5 <- plot_logo_region(seq_path = seq_path_CsgB, region = CsgB_regions[["R5"]])

plot_grid(A1, A2, A3, A4, A5, labels = c("R1", "R2", "R3", "R4", "R5"), nrow = 3, align = "h")

plot_grid(B1, B2, B3, B4, B5, labels = c("R1", "R2", "R3", "R4", "R5"), nrow = 3, align = "h")

@
  

Both CsgA and CsgB are characterized by a regional structure of five repeated motifs. We found out that the general motif (S-X5-Q-X-G-X2-N-X-A-X3-Q) (the serin in absent in the case of CsgB) is faithfully preserved among different variants of CsgA and CsgB. The residual variability in motifs of both proteins does not affect the sequence of other protein.

The opposite situation happens in CsgC, where mutations are not limited to specific motifs. On the other hand, CsgA and CsgC keep a similar level of the sequence divergence aside from species of Enterobacter and Salmonella genera, where CsgC accumulates more mutations.


    \end{block}
    \vfill
}
\end{minipage}
\end{beamercolorbox}
\end{column}


%new column ------------------------------------------------------    

\begin{column}{.51\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}  
\parbox[t][\columnheight]{\textwidth}
{


    \begin{block}{Stability of CsgA curli motif}
      
<<echo = FALSE, message=FALSE,warning=FALSE,fig.width=14,fig.height=10>>=
load("mutation_df.RData")

group_by(mutation_df, pos_max_mutations, nonmutated, max_mutations) %>% 
  summarise(count = length(pos_max_mutations)) %>%  
  ungroup %>% 
  mutate(pos_max_mutations = factor(pos_max_mutations, 
                                    labels = c("S(1)", "Q(7)", "G(9)", "G(11)", "N(12)", "A(14)", "Q(18)")),
         nonmutated = paste0(nonmutated, " nonmutated resiues in the motif"),
         max_mutations = max_mutations - 1) %>% 
  ggplot(aes(x = factor(pos_max_mutations), y = count, color = factor(max_mutations), label = count)) +
  geom_point(size = 4) +
  geom_text_repel(color = "black", force = 10) +
  scale_color_discrete("Number of other amino acids\non a given position") + 
  guides(colour = guide_legend(nrow = 1)) +
  scale_x_discrete("Position in the motif with the largest number of mutations") +
  scale_y_continuous("Number of strains") +
  facet_wrap(~ nonmutated) +
  theme_bw(base_size = 20) +
  theme(legend.position = c(0.85, 0.2))
@
      
Among 2678 investigated strains, 
      
      \end{block}
      \vfill 
      \begin{block}{Conclusions}
      
      The interplay of CsgA, CsgB and CsgC suggests that if a mutation occurs in the region responsible for protein interaction, it should be compensated by mutations in other two proteins. For example, the presence of mutations that decrease the aggregation rate of CsgA, allows CsgC to become a less effective inhibitor without affecting the wellbeing of bacteria. Our preliminary study partially confirms this hypothesis by finding out that genera with less efficient CsgA, as Salmonella, can accumulate more mutations in the sequence of CsgC.

We have not identified any simultaneous mutations between CsgA and CsgB. This may be due that single mutation in one region is not enough to change the protein function and to cause mutations in another protein. Probably, the compensation of single mutations by the regional structure of mentioned proteins is sufficient.

    \end{block}
    \vfill 
    

}
\end{minipage}
\end{beamercolorbox}
\end{column}
\end{columns}  
\end{frame}
\end{document}
